src/daemon/config.js

```diff
@@ -52,11 +52,53 @@ function applyDefaults (ipfsd) {
 
   config.Discovery = config.Discovery || {}
   config.Discovery.MDNS = config.Discovery.MDNS || {}
-  config.Discovery.MDNS.enabled = true
+  config.Discovery.MDNS.Enabled = true
 
   writeConfigFile(ipfsd, config)
 }
 
+// Apply one-time updates to the config of IPFS node.
+// This is the place where we execute fixes and performance tweaks for existing users.
+function migrateConfig (ipfsd) {
+  // Bump revision number when new migration rule is added
+  const REVISION = 1
+  const REVISION_KEY = 'daemonConfigRevision'
+
+  // Migration is applied only once per revision
+  if (store.get(REVISION_KEY) >= REVISION) return
+
+  // Read config
+  let config = null
+  let changed = false
+  try {
+    config = readConfigFile(ipfsd)
+  } catch (err) {
+    // This is a best effort check, dont blow up here, that should happen else where.
+    logger.error(`[daemon] migrateConfig: error reading config file: ${err.message || err}`)
+    return
+  }
+
+  // Cleanup https://github.com/ipfs-shipyard/ipfs-desktop/issues/1631
+  if (config.Discovery && config.Discovery.MDNS && config.Discovery.MDNS.enabled) {
+    config.Discovery.MDNS.Enabled = config.Discovery.MDNS.Enabled || true
```

Comment:
Isn't this `true` everytime?

Reply:
no, user could remove it or explicitly set it to false to disable noise in LAN